src/endpoints/role/role.service.ts

```diff
@@ -41,4 +42,23 @@ export class RoleService {
     });
     return permissions;
   }
+
+  async assignPermissions(request: AssignPermissions) {
+    const isExist = await this.roleRepository.orm.findOneBy({ id: request.roleId });
+    if (!isExist) {
+      throw new HttpException(
+        'ERROR.NOT_FOUND',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    await this.rolePermissionRepository.orm.delete({ roleId: request.roleId });
+    if (request.permissionIds?.length > 0) {
+      const rolePermissions = request.permissionIds.map(permissionId => {
+        return { id: uuid(), roleId: request.roleId, permissionId: permissionId };
+      });
+
+      await this.rolePermissionRepository.orm.insert(rolePermissions);
+    }
+  }
```

Comment:
For this, please use a transaction here.

Look, https://docs.nestjs.com/techniques/database#typeorm-transactions